wagtail/admin/forms/account.py

```diff
@@ -115,7 +122,55 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._original_avatar = self.instance.avatar
 
+    @contextmanager
+    def create_avatar_context(self, avatar):
+        avatar_context = willow.Image.open(avatar)
+        yield avatar_context
+
+    def reduce_avatar(self, avatar):
+        default_avatar_size = (400, 400)
+
+        with self.create_avatar_context(avatar) as willow_image:
+            width, height = willow_image.get_size()
+
+            should_reduce_avatar = (
+                width > default_avatar_size[0] or height > default_avatar_size[1]
+            )
+
+            if should_reduce_avatar:
+                temp_buffer = BytesIO()
```

Comment:
Issue: Resizing images can be quite large. I'd recommend using `SpooledTemporaryFile` here, as is done when resizing renditions:

https://github.com/wagtail/wagtail/blob/9889bddcb8823369f7a26c6b6f4a3eb6784d5985/wagtail/images/views/images.py#L296